### PR TITLE
We need to remove overlay as an option in containers/storage

### PR DESCRIPTION
**Pull request by rhatdan**
Since we prefer our users to use overlay2, there is not reason to
support and provide overlay as an option, and potentially use a worse
driver by accident.

We also need to make overlay2 the prefered driver in containers/storage.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@rhvgoyal @nalind PTAL

**Comment by rhatdan** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=1435767

**Comment by rhatdan** [comment]
@jeremyeder PTAL

File: drivers/driver_linux.go [hunk 1]
```diff
@@ -53,11 +53,11 @@ const (
 var (
 	// Slice of drivers that should be used in an order
 	priority = []string{
+		"overlay",
```
Comment (jeremyeder) [inline drivers/driver_linux.go]: This means we default to overlay in containers/storage, and overlay2-behavior is now default. 

Do you foresee any possibility to misunderstand this, and people ask "where did overlay2 go in containers/storage?"

Should we have a similar proposal for docker in that regard?

Comment (rhatdan) [inline drivers/driver_linux.go]: There was no overlay2 in the list, and I am no expert on containers/storage.  My goal with this change is

Comment (nalind) [inline drivers/driver_linux.go]: This should be "overlay2", I think.

File: drivers/register/register_overlay.go [hunk 2]
```diff
@@ -1,9 +0,0 @@
-// +build !exclude_graphdriver_overlay,linux
-
-package register
-
-import (
-	// register the overlay graphdriver
-	_ "github.com/containers/storage/drivers/overlay"
```
Comment (nalind) [inline drivers/register/register_overlay.go]: We need to drop this line, but not the rest of this file.

**Comment by nalind** [comment]
LGTM.

**Comment by rhatdan** [comment]
@nalind I can't merge.

**Pull request merged**